openedx/core/djangoapps/content/learning_sequences/api/processors/special_exams.py

```diff
@@ -54,26 +57,49 @@ def exam_data(self, pruned_course_outline: UserCourseOutlineData) -> SpecialExam
                         continue
 
                     special_exam_attempt_context = None
-                    try:
-                        # Calls into edx_proctoring subsystem to get relevant special exam information.
-                        # This will return None, if (user, course_id, content_id) is not applicable.
-                        special_exam_attempt_context = get_attempt_status_summary(
-                            self.user.id,
-                            str(self.course_key),
-                            str(sequence.usage_key)
-                        )
-                    except ProctoredExamNotFoundException:
-                        log.info(
-                            'No exam found for {sequence_key} in {course_key}'.format(
-                                sequence_key=sequence.usage_key,
-                                course_key=self.course_key
+
+                    # if exam waffle flag enabled, then use exams logic
+                    if exams_ida_enabled(self.course_key):
```

Comment:
Is there a better way to have this logic to set the short description? I can at least move to a helper method as I refactor/cleanup.